scripts/casefold_db_dry_run.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python
```

Comment:
this should be the same file as `casefold_db.py` -- but should be specified with a command-line argument `--dry-run` for example.

Peruse https://docs.python.org/3/library/argparse.html for the built-in way to parse arguments (a very worthwhile thing to know).

I'd be in favour of adding some flag like `--apply` that you have to pass to actually run the full (non-dry) operation, or making it so you have to pass in the path to the config file (perhaps the config file option is better).

This is just my opinion, but: if someone comes along and types just `casefold_db.py`, it shouldn't do anything (why? because it's too easy to do when you don't know what you're doing). `casefold_db.py --apply` or `casefold_db.py path/to/sydent_config.yaml` has a bit more intent and 'I know what I'm doing' behind it.

`synapse_port_db` works this way too:
```
(.venv) rei@lithium:~/work/synapse$ synapse_port_db 
usage: synapse_port_db [-h] [-v] --sqlite-database SQLITE_DATABASE --postgres-config POSTGRES_CONFIG [--curses] [--batch-size BATCH_SIZE]
synapse_port_db: error: the following arguments are required: --sqlite-database, --postgres-config
```
(synapse_port_db might also be a good thing to look at for reference, though have a go using just the docs to start with to see if you can figure it out)